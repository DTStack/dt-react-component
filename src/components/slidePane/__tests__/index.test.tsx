import React from 'react';
import SlidePane from '../index';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('test SlidePane ', () => {
    const expectValues = {
        children: (
            <div>
                <h1>success</h1>
            </div>
        ),
        visible: true,
        style: { color: 'red', background: '#FF7C12' },
    };
    test('should render correct', () => {
        const { container, getByTestId } = render(
            <SlidePane visible={expectValues.visible} style={expectValues.style}>
                {expectValues.children}
            </SlidePane>
        );
        expect(container.firstChild).toHaveStyle(expectValues.style);
        const oDiv = getByTestId('slidepane_container');
        expect(oDiv).not.toBeNull();
        expect(oDiv).toBeVisible();
        expect(oDiv.innerHTML).toEqual('<div><h1>success</h1></div>');
        expect(container).toMatchSnapshot();
    });
    test('should be invisible', () => {
        const { getByTestId } = render(
            <SlidePane visible={false}>{expectValues.children}</SlidePane>
        );
        const oDiv = getByTestId('slidepane_container');
        expect(oDiv).not.toBeVisible();
    });
});
