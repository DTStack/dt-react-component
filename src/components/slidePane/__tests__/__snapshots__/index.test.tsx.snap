// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SlidePane  should render correct 1`] = `
<div>
  <div
    class="dtc-slide-pane"
    style="top: 0px; color: red; background: rgb(255, 124, 18);"
  >
    <div
      class="dtc-slide-pane-content"
      data-testid="slidepane_container"
      style="display: block; height: 100%;"
    >
      <div>
        <h1>
          success
        </h1>
      </div>
    </div>
    <img
      alt=""
      class="dtc-slide-pane-icon"
      data-testid="slidepane_action"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAABwCAYAAAAE0LDPAAAAAXNSR0IArs4c6QAAAnlJREFUaEPtmk9IVHEQx78jufpw7ykouuEaBG6dEmrXOmRRt476VgvyEngKlNqQMu3PoSU7iCBq4CW2olNdRMRr7HasnnZb0y57WGODzUDjt/RIonbeyzfQYd51h/m8+c7395u3MOSs5Xexr4dKRLt5AhYJtbPRaNO7velo/4Bf6YhoB8B0R3vzNSLaNr8ECnBRRLTc0d583kBEAJU3J5o6HG0ZkgTs1OBATAxgqqghTIoCiPBeFABQSRggZFOxg/anC0ElYq9JlUglYhVgA9RFKhGrABugLlKJWAXYAHWRSsQqwAaoi1QiVgE2gHVRejKDUKgWQ1cvmj/XbMLfA3jA42dwVvPo7+tBd/xo8ICNzQLG7y3AskIYvz2IcEO9LwhbgcmWebGCpeUcEvEYBvrOBg8ol7cxOjaP4lYJqZEkIm2NniGeKjDZ3mQ/YPbJa7S1NiI1YntuuGeAgTx8lMHqx3XYvWdwOnHMUxW+AJufC7hzdwH1VggTt64gHLZYiC+Ayfb85QoWl3KIn+jEpeS54AHlb98xOjaHYrGEG8M2DkWaqkJ8V2CyZd86mJl7hdaWg7h5PVm14f8EMJD0zxM+ePkCuo4f+WsV/ycgm3MwMy8k0d4mp4ZtRIJusmvTxMlODNgB21T8oLlXRbK3B6cS3maDZxeJXnbudb315Wvl9AZ+XbsDpzseQ3/QA+fTRgET9wVHpvmqcNbWK6PSjEy/D9vkB+mnaLDq5D5b/L6x7+8iBbBNVolUItYDKpFKxCrABqiLVCJWATZAXaQSsQqwAeoilYhVgA1QFzESCS8wia9giS6Rma1B0TU40UU+sVVEoWXK6uugPwBjVkdxauLcgwAAAABJRU5ErkJggg=="
    />
  </div>
</div>
`;
